src/server/server.js

```diff
@@ -19,22 +19,50 @@ import downloadRouter from './download';
 import CodedError from './utils/error';
 import { statusRouter, versionRouter } from './endpoints';
 
+let server;
 const app = express();
 app.use(cors());
 app.use(helmet());
 app.use(bodyParser.json({ limit: '50mb' }));
 
+const refreshRouter  = async (req, res, next) => {
+  res.setHeader('Content-Type', 'application/json; charset=utf-8');
+  try {
+    if (config.allowRefresh) {
+      log.debug('[Refresh] ', JSON.stringify(req.body, null, 4));
+      const jwt = headerParser.parseJWT(req);
+      if (!jwt) {
+        const noJwtError = new CodedError(401, '[Refresh] no JWT user token provided to _refresh function');
+        throw noJwtError;
+      }
+      const authHelper = await getAuthHelperInstance(jwt);
+      console.log("AUTH HELPER: ", authHelper)
+      if (authHelper._accessibleCreateResourceList === undefined || authHelper._accessibleCreateResourceList.length === 0) {
+        const noPermsUser = new CodedError(401, '[Refresh] User cannot refresh Guppy without a valid token that has read access to at least one project');
+        throw noPermsUser;
+      }
+      await server.stop()
+      await initializeAndStartServer();
```

Comment:
This solution is generally ok, but it only works if there is only 1 guppy pod/container/server in the system.
In the cases of how Gen3 is usually deployed, which is on a k8s cluster with a load balancer / reverse proxy in front of pods, and also if there are replicas of guppy pods. This solution will only triggers 1 guppy pod to restart each time, which will causes a discrepancy in the states of pods

Reply:
@mfshao thanks for all the feedback.   In the case of replica guppy pods, is there a link you could provide for how requests from the clients are distributed to the guppy replicas?  In other words, is there a nginx or helm configuration that fans out the requests?

Reply:
@mfshao Alternatively, guppy could checking mappings on a configurable interval and initiate a restart if the version changed.  For example, for index `gen3.aced.io_file_0`

``` curl  localhost:9200/gen3.aced.io_file_0  | jq '.["gen3.aced.io_file_0"].settings.index.version.created '
>>> "7170399"

```